tests/Microsoft.Data.DataFrame.Tests/DataFrameTests.cs

```diff
@@ -0,0 +1,362 @@
+            Data.DataFrame dataFrame = new Data.DataFrame();
+            dataFrame.InsertColumn(0, dataFrameColumn1);
+            dataFrame.InsertColumn(1, dataFrameColumn2);
+            return dataFrame;
+        }
+
+        static public DataFrame MakeDataFrameWithAllColumnTypes(int length)
+        {
+            DataFrame df = MakeDataFrameWithNumericAndStringColumns(length);
+            BaseColumn boolColumn = new PrimitiveColumn<bool>("Bool", Enumerable.Range(0, length).Select(x => x % 2 == 0));
+            df.InsertColumn(df.ColumnCount, boolColumn);
+            return df;
+        }
+
+        static public DataFrame MakeDataFrameWithNumericAndStringColumns(int length)
+        {
+            DataFrame df = MakeDataFrameWithNumericColumns(length);
+            BaseColumn stringColumn = new StringColumn("String", Enumerable.Range(0, length).Select(x => x.ToString()));
+            df.InsertColumn(df.ColumnCount, stringColumn);
+            return df;
+        }
+
+        static public DataFrame MakeDataFrameWithNumericColumns(int length)
+        {
+            BaseColumn byteColumn = new PrimitiveColumn<byte>("Byte", Enumerable.Range(0, length).Select(x => (byte)x));
+            BaseColumn charColumn = new PrimitiveColumn<char>("Char", Enumerable.Range(0, length).Select(x => (char)x));
+            BaseColumn decimalColumn = new PrimitiveColumn<decimal>("Decimal", Enumerable.Range(0, length).Select(x => (decimal)x));
+            BaseColumn doubleColumn = new PrimitiveColumn<double>("Double", Enumerable.Range(0, length).Select(x => (double)x));
+            BaseColumn floatColumn = new PrimitiveColumn<float>("Float", Enumerable.Range(0, length).Select(x => (float)x));
+            BaseColumn intColumn = new PrimitiveColumn<int>("Int", Enumerable.Range(0, length).Select(x => x));
+            BaseColumn longColumn = new PrimitiveColumn<long>("Long", Enumerable.Range(0, length).Select(x => (long)x));
+            BaseColumn sbyteColumn = new PrimitiveColumn<sbyte>("Sbyte", Enumerable.Range(0, length).Select(x => (sbyte)x));
+            BaseColumn shortColumn = new PrimitiveColumn<short>("Short", Enumerable.Range(0, length).Select(x => (short)x));
+            BaseColumn uintColumn = new PrimitiveColumn<uint>("Uint", Enumerable.Range(0, length).Select(x => (uint)x));
+            BaseColumn ulongColumn = new PrimitiveColumn<ulong>("Ulong", Enumerable.Range(0, length).Select(x => (ulong)x));
+            BaseColumn ushortColumn = new PrimitiveColumn<ushort>("Ushort", Enumerable.Range(0, length).Select(x => (ushort)x));
+
+            DataFrame dataFrame = new DataFrame(new List<BaseColumn> { byteColumn, charColumn, decimalColumn, doubleColumn, floatColumn, intColumn, longColumn, sbyteColumn, shortColumn, uintColumn, ulongColumn, ushortColumn});
+
+            return dataFrame;
+        }
+
+        [Fact]
+        public void TestIndexer()
```

Comment:
We need to write some more tests. I don't think these tests cover the amount of code that is in the library.